plugin-unmanaged-racks/redfish/error.go

```diff
@@ -0,0 +1,221 @@
+	ResourceAlreadyExists = "Base.1.6.1.ResourceAlreadyExists"
+	// ActionParameterNotSupported indicates that the parameter supplied for the action is not supported on the resource.
+	ActionParameterNotSupported = "Base.1.6.1.ActionParameterNotSupported"
+	// ResourceCannotBeDeleted indicates the requested delete operation cannot be performed
+	ResourceCannotBeDeleted = "Base.1.6.1.ResourceCannotBeDeleted"
+	// PropertyValueConflict indicates that the requested write of a property value could not be completed, because of a conflict with another property value.
+	PropertyValueConflict = "Base.1.6.1.PropertyValueConflict"
+)
+
+//CommonError struct definition
+type CommonError struct {
+	Error ErrorClass `json:"error"`
+}
+
+//ErrorClass struct definition
+type ErrorClass struct {
+	Code                string            `json:"code"`
+	Message             string            `json:"message"`
+	MessageExtendedInfo []MsgExtendedInfo `json:"@Message.ExtendedInfo,omitempty"`
+}
+
+//MsgExtendedInfo struct definition
+type MsgExtendedInfo struct {
+	OdataType   string        `json:"@odata.type,omitempty"`
+	MessageID   string        `json:"MessageId,omitempty"`
+	Message     string        `json:"Message,omitempty"`
+	MessageArgs []interface{} `json:"MessageArgs,omitempty"`
+	Severity    string        `json:"Severity,omitempty"`
+	Resolution  string        `json:"Resolution,omitempty"`
+}
+
+// CreateError creates new instance of CommonError
+func CreateError(code string, errorMessage string) CommonError {
+	return CommonError{
+		Error: ErrorClass{
+			Code:    code,
+			Message: errorMessage,
+		},
+	}
+}
+
+// NewError creates new instance of CommonError
+func NewError(msgs ...MsgExtendedInfo) *CommonError {
+	return &CommonError{
+		Error: ErrorClass{
+			Code:                GeneralError,
+			Message:             "An error has occurred. See ExtendedInfo for more information.",
+			MessageExtendedInfo: msgs,
+		},
+	}
+}
+
+// AddExtendedInfo adds instance of MsgExtendedInfo to `CommonError.Error.MessageExtendedInfo` collection
+func (e *CommonError) AddExtendedInfo(ei MsgExtendedInfo) *CommonError {
+	e.Error.MessageExtendedInfo = append(e.Error.MessageExtendedInfo, ei)
+	return e
+}
+
+// NewMalformedJSONMsg constructs instance of MsgExtendedInfo representing `MalformedJSON` error message
+func NewMalformedJSONMsg(errMsg string) MsgExtendedInfo {
+	return MsgExtendedInfo{
+		OdataType:  "#Message.v1_0_8.Message",
+		MessageID:  MalformedJSON,
+		Message:    "The request body submitted was malformed JSON and could not be parsed by the receiving service: " + errMsg,
+		Severity:   "Critical",
+		Resolution: "Ensure that the request body is valid JSON and resubmit the request.",
+	}
+}
+
+// NewPropertyMissingMsg constructs instance of MsgExtendedInfo representing `PropertyMissing` error message
+func NewPropertyMissingMsg(missingProperty, errorMsg string) MsgExtendedInfo {
```

Comment:
There is a similar implementation for generating errors in lib-utilities. IMHO the same can be reused here.